src/iceberg_functions/iceberg_multi_file_reader.cpp

```diff
@@ -312,7 +311,8 @@ void IcebergMultiFileReader::FinalizeBind(MultiFileReaderData &reader_data, cons
 	auto &data_file = multi_file_list.data_files[file_id];
 
 	// The path of the data file where this chunk was read from
-	auto &file_path = data_file.file_path;
+	// Copy the file path otherwise it gets overwritten somewhere
+	const auto file_path = data_file.file_path;
```

Comment:
This sounds wrong to me, that would indicate that the MultiFileList is either not const, or not living long enough (if it gets destroyed before we use the data_file referenced here)